src/context/AppProvider/hooks/useDiscoverAccounts.tsx

```diff
@@ -0,0 +1,138 @@
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
+import { MetaMaskMultiChainHDWallet } from '@shapeshiftoss/hdwallet-metamask-multichain'
+import type { AccountMetadataById } from '@shapeshiftoss/types'
+import { useQueries } from '@tanstack/react-query'
+import { useMemo } from 'react'
+
+import { getAccountIdsWithActivityAndMetadata } from '@/components/ManageAccountsDrawer/helpers'
+import { usePlugins } from '@/context/PluginProvider/PluginProvider'
+import { useIsSnapInstalled } from '@/hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useWallet } from '@/hooks/useWallet/useWallet'
+import { walletSupportsChain } from '@/hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { METAMASK_RDNS } from '@/lib/mipd'
+import { selectWalletRdns } from '@/state/slices/localWalletSlice/selectors'
+import { portfolio } from '@/state/slices/portfolioSlice/portfolioSlice'
+import { store, useAppDispatch, useAppSelector } from '@/state/store'
+
+export const useDiscoverAccounts = () => {
+  const dispatch = useAppDispatch()
+  const { isSnapInstalled } = useIsSnapInstalled()
+  const { deviceId, wallet } = useWallet().state
+  const { supportedChains } = usePlugins()
+  const connectedRdns = useAppSelector(selectWalletRdns)
+
+  const supportedChainIds = useMemo(() => {
+    return supportedChains.filter(chainId =>
+      walletSupportsChain({ chainId, wallet, isSnapInstalled, checkConnectedAccountIds: false }),
+    )
+  }, [supportedChains, wallet, isSnapInstalled])
+
+  const queries = useMemo(
+    () =>
+      supportedChainIds.map(chainId => ({
+        queryKey: [
+          'useDiscoverAccounts',
+          {
+            deviceId,
+            isSnapInstalled,
+          },
+          chainId,
+        ],
+        queryFn: async () => {
+          const isMetaMaskMultichainWallet = wallet instanceof MetaMaskMultiChainHDWallet
+
+          if (
+            !wallet ||
+            isLedger(wallet) ||
+            // Before connecting to MetaMask, isSnapInstalled is null then switch to false when the hook reacts, we would run the discovery 2 times
+            (connectedRdns === METAMASK_RDNS && isSnapInstalled === null)
+          ) {
+            return { accountMetadataByAccountId: {}, hasActivity: false }
+          }
+
+          const walletId = await wallet.getDeviceID()
+          const isMultiAccountWallet = wallet.supportsBip44Accounts()
+          const currentPortfolio = portfolio.selectors.selectPortfolio(store.getState())
+
+          let accountNumber = 0
+          let hasActivity = true
+          const chainAccountMetadata: AccountMetadataById = {}
+
+          while (hasActivity) {
+            if (accountNumber > 0) {
+              if (!isMultiAccountWallet) break
+              if (isMetaMaskMultichainWallet && !isSnapInstalled) break
+            }
+
+            try {
+              const accountIdWithActivityAndMetadata = await getAccountIdsWithActivityAndMetadata(
```

Comment:
q: there are still 5 occurrences of `getAccount.initiate()`, is this expected? 

<img width="1342" height="127" alt="image" src="https://github.com/user-attachments/assets/3cb6e698-e176-4d8c-bf5c-26cfd7833994" />

This effectively means that `adapter.getAccount(pubkey)` will be called again in those places - `usePortfolioFetch()` in particular seems like it would be an effective duplicate.

Though, rubber-ducking myself, `getAccount()` query itself uses `accountManagement.getAccount(accountId)` so *should* be sane